### PR TITLE
[FLINK-20596] Remove Scheduler

### PR DESCRIPTION
**Pull request by XComp**
## What is the purpose of the change

This PR removes code related to the `Scheduler` interface as part of a major cleanup in [FLINK-20589](https://issues.apache.org/jira/browse/FLINK-20589).

## Brief change log

* Removed `Scheduler` and `SlotProvider` and related classes

## Verifying this change

* the build pipeline should still succeed

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers:  no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/SlotRequestId.java [hunk 1]
```diff
@@ -18,14 +18,15 @@
...
+import org.apache.flink.runtime.jobmaster.slotpool.PhysicalSlotProvider;
 import org.apache.flink.runtime.jobmaster.slotpool.SlotPool;
-import org.apache.flink.runtime.jobmaster.slotpool.SlotProvider;
 import org.apache.flink.util.AbstractID;
 
 /**
- * This ID identifies the request for a logical slot from the Execution to the {@link SlotPool} oe
- * {@link SlotProvider}. The logical slot may be a physical slot or a sub-slot thereof, in the case
- * of slot sharing.
+ * This ID identifies the request for a slot from the Execution to the {@link SlotPool} oe {@link
```
Comment (rmetzger) [inline flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/SlotRequestId.java:27]:
```suggestion
 * This ID identifies the request for a slot from the Execution to the {@link SlotPool} or {@link
```

**Review by rmetzger — APPROVED** [review]
Thanks a lot for the removal. 
The changes seem reasonable. I also checked the git log for the reasoning of the test removal / changes.

As a side note, some committers are pretty opinionated (not me) about the commit message format: https://www.robertcooper.me/git-commit-messages. I personally try to follow the line-length limit on a best-effort basis.

**Comment by XComp** [comment]
Thanks for the remarks and sharing the link, @rmetzger . I addressed your comments and, in addition to that, deleted `SlotAndLocality.java` as it's not used anymore. The changes have been squashed and are ready to be merged.

**Comment by rmetzger** [comment]
Thanks for addressing my comments. Merging ...

**Pull request merged**